src/panoramic/cli/field/client.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from typing import Any, Dict, List, Optional
+from urllib.parse import urljoin
+
+from panoramic.auth import OAuth2Client
+from panoramic.cli.clients import VersionedClient
+from panoramic.cli.config.auth import get_client_id, get_client_secret
+from panoramic.cli.config.field import get_base_url
+
+logger = logging.getLogger(__name__)
+
+
+class Field:
+    def __init__(
+        self,
+        *,
+        slug: str,
+        group: str,
+        display_name: str,
+        data_type: str,
+        field_type: str,
+        description: Optional[str],
+        data_source: Optional[str],
+        calculation: Optional[str],
+        aggregation: Optional[Dict[str, Any]],
+        display_format: Optional[str],
+    ):
+        self.slug = slug
+        self.group = group
+        self.display_name = display_name
+        self.data_type = data_type
+        self.field_type = field_type
+        self.description = description
+        self.data_source = data_source
+        self.calculation = calculation
+        self.aggregation = aggregation
+        self.display_format = display_format
+
+    @classmethod
+    def from_dict(cls, inputs: Dict[str, Any]) -> 'Field':
+        return cls(
+            slug=inputs['slug'],
+            group=inputs['group'],
+            display_name=inputs['display_name'],
+            data_type=inputs['data_type'],
+            field_type=inputs['field_type'],
+            description=inputs.get('description'),
+            data_source=inputs.get('data_source'),
+            aggregation=inputs.get('aggregation'),
+            calculation=inputs.get('calculation'),
+            display_format=inputs.get('display_format'),
+        )
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            'slug': self.slug,
+            'group': self.group,
+            'display_name': self.display_name,
+            'data_type': self.data_type,
+            'field_type': self.field_type,
+            'description': self.description,
+            'calculation': self.calculation,
+            'aggregation': self.aggregation,
+            'data_source': self.data_source,
+            'display_format': self.display_format,
+        }
+
+    def __hash__(self) -> int:
+        return hash(self.to_dict())
```

Comment:
Dict is not hashable so this doesn't work

Reply:
You're right - also problem for all the other remote model classes https://github.com/panoramichq/panoramic-cli/blob/fd7bcf23cd32bffe97f3aa1cfb4678f58a872686/src/panoramic/cli/model/client.py#L139 (thought we already fixed that?)

Reply:
Let's fix it in a follow-up for all the affected classes.

Reply:
So did I... ok let's create a bug ticket and fix there